### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by ellehallal**
PlayerMaker
- Rename to PlayerFactory

PlayerSelector
- Rename to PlayerValidator

Board: 
- change winning_lines class variable to constant
- Delete clear_squares
- Change available_squares_quantity to total_available_squares and make private
- Refactor available_squares

Controller:
- Remove squares parameter from main_game

Display:
- amend ask_for_player_selection. 
- Make methods related to displaying outcome private
- Amend remove new lines from show_computer_thinking

Game Manager: 
- amend order of input for test
- add welcome message to play method
- make play_again_private

Game: 
- make winning_player and toggle_current_player private
- test player 2's mark is returned when they have a winning line

Game Factory
- amend create_game

Remove duplicated methods and tests. Add required classes in text

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class Board
-  @@winning_lines = [
+  WINNING_LINES = [
     [0, 1, 2],
     [3, 4, 5],
     [6, 7, 8],
```
Comment (Kotauror) [inline lib/board.rb:5]: Using const - Awesome decision! 👏

File: lib/board.rb [hunk 2]
```diff
@@ -36,22 +36,16 @@ def winning_player_exists?(player1_mark, player2_mark)
...
+  private
 
-  def available_squares
-    available = []
-    @squares.each do |square|
-      available << square if square.is_a? Integer
-    end
-    available
+  def total_available_squares
+    available_squares.size
```
Comment (Kotauror) [inline lib/board.rb:49]: Good refactoring above! I love how the code is getting cleaner and better with every PR! :)

File: lib/game.rb [hunk 3]
```diff
@@ -18,30 +36,11 @@ def winning_player
...
-    @board.player_make_move(@current_player.mark, player_move)
-  end
-
   def toggle_current_player
     @current_player = if @current_player == @player1
                         @player2
                       else
                         @player1
                       end
   end
```
Comment (Kotauror) [inline lib/game.rb:45]: I know it's a pain, but you need to do something about this indentation.

File: spec/player_factory_spec.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
+require 'human_player'
+require 'computer_player'
+require 'player_factory'
+require 'display'
+
+RSpec.describe PlayerFactory do
```
Comment (Kotauror) [inline spec/player_factory_spec.rb:6]: A general design / naming question - it sounds (to me) a bit bizarre that PlayerValidator has a PlayerFactory, not the other way around. What do you think about it?

File: lib/board.rb (hunk 1 above)
Comment (ellehallal) [reply lib/board.rb:5]: Thanks!

File: lib/board.rb (hunk 2 above)
Comment (ellehallal) [reply lib/board.rb:49]: Thanks! This was Matt's suggestion :)

File: spec/player_factory_spec.rb (hunk 4 above)
Comment (ellehallal) [reply spec/player_factory_spec.rb:6]: That's a good point. I'll look into this, thanks!

**Pull request merged**